### ISSUE TITLE
DayPicker Improvement

### ISSUE THREAD

**Issue by jossmac** [keystonejs/keystone#215]
The `DayPicker` component and corresponding `CalendarDay` field are basically functional, but require a bit more work to be complete:

#### Aesthetic

- Moved to new issue: range-selection start/end day

#### Functionality

- [x] range-selection
- [x] selectable month from native `<select/>`
- [x] selectable year from native `<select/>`
- [ ] unavailable/disabled days (greyed out)
- [x] select year by typing (if the list of available years is too long for a `<select/>`

#### A11y

- [ ] keyboard navigation
- [ ] screen-reader audit and subsequent updates

#### Ideas

- [ ] i18n support for month/weekday labels

**Comment by MadeByMike** [comment]
@molomby can you add some notes about the bug with how the dates are stored?

**Comment by molomby** [comment]
@MadeByMike I don't have any problem with how `CalendarDay` values are stored; you might be thinking of my work on the [`DateTimeUtc` type](https://github.com/keystonejs/keystone-5/pull/1414) 🙂

**Comment by jossmac** [comment]
React day picker should release [version 8](https://github.com/gpbl/react-day-picker/issues/942), which includes an improved API and hooks support, sometime soon. I'm happy to pick up this issues once that lands.
